lib/features/overview/pages/overview_page.dart

```diff
@@ -290,6 +335,104 @@ class OverviewPage extends StatelessWidget {
     );
   }
 
+  Widget _buildAppBarItem({
+    required GivtState state,
+    required BuildContext context,
+    required Widget child,
+    required Icon icon,
+    Color? color,
+  }) {
+    return Visibility(
+      visible: state.givtGroups.isNotEmpty,
+      child: IconButton(
+        icon: icon,
+        onPressed: () => showModalBottomSheet<void>(
+          context: context,
+          showDragHandle: true,
+          isScrollControlled: true,
+          useSafeArea: true,
+          backgroundColor: color ?? AppTheme.givtBlue,
+          builder: (context) =>
+              Container(padding: const EdgeInsets.all(20), child: child),
+        ),
+      ),
+    );
+  }
+
+  Widget _buildAnnualOverviewHeader(
+      {required String text, required String email}) {
+    return RichText(
+      textAlign: TextAlign.center,
+      text: TextSpan(
+        text: text,
+        style: const TextStyle(
+          fontSize: 16,
+          color: AppTheme.givtBlue,
+          fontWeight: FontWeight.normal,
+        ),
+        children: <TextSpan>[
+          TextSpan(
+            text: ' $email',
+            style: const TextStyle(
+              fontWeight: FontWeight.bold,
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  Widget _buildYearDropdown(GivtState state, BuildContext context, Size size) {
+    return DropdownButtonFormField<String>(
+      validator: (value) {
+        if (value == null) {
+          return '';
+        }
+
+        return null;
+      },
+      value: _getYears(state).contains(_overviewYearController)
+          ? _overviewYearController
+          : _getYears(state).first,
+      decoration: const InputDecoration(
+        contentPadding: EdgeInsets.symmetric(
+          horizontal: 10,
+          vertical: 5,
+        ),
+        enabledBorder: UnderlineInputBorder(
+          borderSide: BorderSide(
+            color: AppTheme.givtBlue,
+          ),
+        ),
+        focusedBorder: UnderlineInputBorder(
+          borderSide: BorderSide(
+            color: AppTheme.givtBlue,
+          ),
+        ),
+        errorStyle: TextStyle(
+          height: 0,
+        ),
+      ),
+      menuMaxHeight: size.height * 0.3,
+      items: _getYears(state)
+          .map(
+            (String country) => DropdownMenuItem(
+              value: country,
+              child: Text(
+                country,
+                style: Theme.of(context).textTheme.bodyLarge,
+              ),
+            ),
+          )
+          .toList(),
+      onChanged: (String? newValue) {
+        setState(() {
+          _overviewYearController = newValue!;
+        });
+      },
+    );
+  }
```

Comment:
Can't you extract this into a separate widget and use a callback to get the selected year? In this way you don't need to convert the whole widget into a stateful one

Reply:
Done